logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java

```diff
@@ -219,8 +235,64 @@ void dropTailSegment() throws IOException {
         final Path beheadedSegment = oldestSegment.get();
         final long segmentSize = Files.size(beheadedSegment);
         currentQueueSize.add(-segmentSize);
-        Files.delete(beheadedSegment);
-        logger.debug("Deleted exceeded retained size segment file {}", beheadedSegment);
+        try {
+            final long deletedEvents = countEventsInSegment(beheadedSegment);
+            droppedEvents.add(deletedEvents);
+        } finally {
+            Files.delete(beheadedSegment);
+        }
+        lastError = String.format("Deleted exceeded retained size segment file %s", beheadedSegment);
```

Comment:
I'm not sure I would consider this an error, as it will happen in the normal operation of the DLQ, rather this should remain as a log entry

Reply:
Logged as a debug